### ISSUE TITLE
How to add my SSL certificate to Microcks for integration tests

### ISSUE THREAD

**Issue by vrakoton** [microcks/microcks#167]
Hi,

I am trying Microcks 0.7.2 on our own Openshift PaaS. I used the Charts provided in the official install docs. 

Is there a way to add my certificates to call my endpoints for the integration tests without rebuilding the official Microcks image?

I can see Secrets are only handling authentication tokens and not certificates for self signed or unkown certification authority. 

Thanks,
Vina.

**Comment by lbroudoux** [comment]
Hi @vrakoton ,

thank you for testing Microcks.

Secrets should also allow you to put put certificates for your own CA. On the screen below, I'm able to assign a certificate in addition of tokens. However they can only be assigned to an import job for pulling repository.

<img width="1788" alt="Capture d’écran 2019-10-21 à 13 27 11" src="https://user-images.githubusercontent.com/1538635/67234256-1b553300-f445-11e9-9a0e-cf3948adf0c0.png">

Currently, in `0.7.2-SNAPSHOT` it appears that HTTPS with unknown CA is not supported for `HTTP` and `OPEN API SCHEMA` contract testing strategies but supported by the `POSTMAN` one.

A first step could be to allow unsecure HTTPS testing before trying to extend the usage of Secrets to integration tests. What do you think of that?

**Comment by vrakoton** [comment]
Hi @lbroudoux,

Thank you for answering this thread.

I am using the Postman runner but I am not 100% percent sure of what you are saying. Let me rephrase what I understood:
- The microcks JVM requires trusted certificates with known CA
- The postman runner nodeJS app does not care about untrusted HTTPS endpoints

Am I right?

This is what I can see in the microcks JVM logs (but I would expect the postman runner to complain about this really... rather than the JVM):

```
[ERROR] io.github.microcks.util.test.HttpTestRunner - IOException while executing request 1 or more results on https://<redacted>
--
  | javax.net.ssl.SSLException: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
  | at sun.security.ssl.Alerts.getSSLException(Alerts.java:208)
  | at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1964)
  | at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1921)
  | at sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1904)
  | at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1420)
  | at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1397)
  | at sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:559)
  | at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185)
  | at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:162)
  | at org.springframework.http.client.SimpleBufferingClientHttpRequest.executeInternal(SimpleBufferingClientHttpRequest.java:78)
  | at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
  | at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:53)
  | at io.github.microcks.util.test.HttpTestRunner.runTest(HttpTestRunner.java:133)
  | at io.github.microcks.util.test.HttpTestRunner.runTest(HttpTestRunner.java:48)
  | at io.github.microcks.service.TestRunnerService.launchTestsInternal(TestRunnerService.java:118)
  | at io.github.microcks.service.TestRunnerService$$FastClassBySpringCGLIB$$e3e9c696.invoke(<generated>)
  | at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
  | at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:736)
  | at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
  | at org.springframework.aop.interceptor.AsyncExecutionInterceptor$1.call(AsyncExecutionInterceptor.java:115)
  | at org.springframework.aop.interceptor.AsyncExecutionAspectSupport$CompletableFutureDelegate$1.get(AsyncExecutionAspectSupport.java:328)
  | at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
  | at java.lang.Thread.run(Thread.java:748)
  | Caused by: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
  | at sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:91)
  | at sun.security.validator.Validator.getInstance(Validator.java:181)
  | at sun.security.ssl.X509TrustManagerImpl.getValidator(X509TrustManagerImpl.java:312)
  | at sun.security.ssl.X509TrustManagerImpl.checkTrustedInit(X509TrustManagerImpl.java:171)
  | at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:184)
  | at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)
  | at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1596)
  | at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)
  | at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1052)
  | at sun.security.ssl.Handshaker.process_record(Handshaker.java:987)
  | at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1072)
  | at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1385)
  | at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1413)
  | ... 18 common frames omitted
  | Caused by: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
  | at java.security.cert.PKIXParameters.setTrustAnchors(PKIXParameters.java:200)
  | at java.security.cert.PKIXParameters.<init>(PKIXParameters.java:120)
  | at java.security.cert.PKIXBuilderParameters.<init>(PKIXBuilderParameters.java:104)
  | at sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:89)
  | ... 30 common frames omitted
```
So you are probably right to say that the straightest path to the resolution of this issue is to add a parameter to instruct the HTTP test runner to use a "no SSL validation" runner factory. Which makes sense to me as most of the time, in lower environments, customers will often use self signed certificates.

**Comment by lbroudoux** [comment]
Now that #167 is resolved this should work as expected. Please feel free to comment and we'll re-open this issue if not the case.
